### PR TITLE
Extract paths C API implementation

### LINKED ISSUE #1891: [ENH] Add extract_bfs_paths to C API
We are migrating to having python calling the C API.  We should add the new `extract_bfs_paths` algorithm to the C API to facilitate the creation of python bindings.

### PR DESCRIPTION
**Pull request by ChuckHastings**
C API implementation for extract paths.

Closes #1891

### CODE REVIEW

File: cpp/src/c_api/array.hpp [hunk 1]
```diff
@@ -41,9 +41,10 @@ struct cugraph_type_erased_device_array_t {
   }
 
   template <typename T>
-  cugraph_type_erased_device_array_t(rmm::device_uvector<T>&& vec, data_type_id_t type)
-    : data_(vec.release()), size_(vec.size()), type_(type)
+  cugraph_type_erased_device_array_t(rmm::device_uvector<T>& vec, data_type_id_t type)
+    : size_(vec.size()), type_(type)
```
Comment (seunghwak) [inline cpp/src/c_api/array.hpp:45]: Isn't it cleaner to just change the order of size_ and data_ in the struct definition? (so : size_(vec.size()), data_(vec.release()) to work?)

**Review by seunghwak — APPROVED** [review]
Looks good to me, I have just one minor comment to consider.

File: cpp/src/c_api/array.hpp (hunk 1 above)
Comment (ChuckHastings) [reply cpp/src/c_api/array.hpp:45]: Done in latest push.

I had tried changing the order of initialization without changing the order of the data in the struct that did not work.  Changing both was the missing trick.  Thanks!

**Review by kaatish — APPROVED** [review]

**Comment by ChuckHastings** [comment]
@gpucibot merge

**Pull request merged**